src/Metadata/Property/PropertyMetadata.php

```diff
@@ -60,7 +60,7 @@ public function getType()
      *
      * @param Type $type
      *
-     * @return self
+     * @return PropertyMetadata
```

Comment:
>I'd think it's also how IDEs etc.

Well, besides finding that unintuitive, IDE do not recognised it either (at least PhpStorm 2016.1)

But if you're finding it being more of an hassle that' fair, I just want to raise that point :)